packages/cdk-assets/lib/private/docker.ts

```diff
@@ -70,7 +70,7 @@ export class Docker {
     await this.execute(['login',
       '--username', credentials.username,
       '--password-stdin',
-      credentials.endpoint], {
+      credentials.endpoint.replace(/^https?:\/\//, '')], {
```

Comment:
```suggestion
      credentials.endpoint.replace(/^https?:\/\/|\/$/g, '')], {
```

Remove end slash too considering that the test data included an end slash in the endpoint.

Reply:
@rosskarchner Tests also need to be fixed in `packages/cdk-assets/test/docker-images.test.ts`

Here is one test that is failing
https://github.com/aws/aws-cdk/blob/ac9901ada20e0bcadcae0e6f59e5c58220328714/packages/cdk-assets/test/docker-images.test.ts#L154

I believe there are four total that use `https`

Reply:
Thanks! I'll take a look at these

Reply:
@Ragnoroct thanks, I'll take a look at these (once I get a working environment to test with)